### ISSUE TITLE
`end` is parameter is not really optional / how to downtime indefinitely?

### ISSUE THREAD

**Issue by felixfriedrich** [DataDog/terraform-provider-datadog#747]
### Terraform Version

v0.13.0

### Affected Resource(s)

-   https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/downtime

### Terraform Configuration Files

```hcl
resource "datadog_downtime" "memory_usage_downtime" {
  # https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/downtime
  for_each   = var.mute_saturation && var.enable_saturation ? var.services : {}
  monitor_id = datadog_monitor.memory_usage_monitor[each.key].id
  scope      = ["*"]
  message    = "This downtime has been created via Terraform. See: https://github.com/deliveryhero/pd-box/tree/master/guidelines/observability/datadog#3-monitoring"
}
```

### Question

I am not sure whether it's a bug, or if I simply have a question. `end` is documented as optional. If I create a downtime with no `end` parameter, next time I run TF though, `end` changes, from some value to `null`.
I know, I can make TF ignore this parameter. However, how do I create a downtime which lasts forever? I tried not using `end` at all, `0` and `-1`, all don't seem to do the trick. I also couldn't find any hint in the docs or simply via Google.

In Datadog's UI this seems to be possible, however I cannot figure out how to translate this into parameters for this TF ressource.

Thanks,
Felix

**Comment by therve** [comment]
Hi,

Not specifying `end` should just work. The state will be set to `0`, but it shouldn't generate a diff. Are you seeing a diff when not setting it? What's your provider version? Thanks.

**Comment by felixfriedrich** [comment]
I looked into what version of the provider I was using. Turned out I had two different version specified, one in the module and one where the module was being used. After sorting out the mess, all worked as expected :-)

Thanks,
Felix

**Comment by felixfriedrich** [comment]
I am afraid I need to open this again. Me using different versions of the provider was a red herring. It did't influence the behaviour. Today I found out what did, when investigating another issues. A related issue, how I found out later.

I use this on my monitors:
```
lifecycle {
   ignore_changes = [silenced]
 }
```

The reason is described here: https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/monitor#silencing-by-hand-and-by-downtimes In short: This code snipped makes is possible to manual mute a monitor without `terraform plan` showing a change like this:

```
~ silenced = {
   - "*" = 0 -> null
}
```

It basically makes TF ignore the mute state of the monitor. According to the docs, the same applies to the TF resource called `terraform_downtime` (https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/downtime). The `datadog_downtime` resource  can be used to mute a monitor and the `silenced` field would also showchanges when running `terraform apply`, if the monitor isn't instructed to ignore changes on the `silenced` field via the code snippet above. So far this is all only information from the docs.
Side note: The `silenced` field is deprecated (see: https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/monitor), such that using the code snipped above is basically the way to go, when using `datadog_downtime` resources.

Now back to the issue I found this morning. Using the code snippet to ignore changes on the `silenced` field and using the `datadog_downtime` resource works fine until the `datadog_downtime` resource is destroyed. `terraform plan` shows that the downtime will be destroyed (I doubled checked the ID) and also actually destroys the resource (I double checked it's gone after plan execution). However the monitor remains muted. After some time I found out that the downtime resource muting the monitor is NOT the one which just has been destroyed. (How could it even? It has been destroyed). The downtime resource now responsible for the monitor being muted  is *new* (I doubled checked it has a different ID from the downtime resource, which just has been destroyed).

I think this happens in order to make this promise true: `The above will make sure that any changes to the silenced attribute are ignored.` (Source: https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/monitor)
In case someone manually mutes a monitor, the code snipped above makes sure that the monitor remains muted, disregarding the value of the (deprecated) `silenced` field. Whether the downtime is now manually created via the UI or automatically via Terraform, doesn't seem to make a difference. The monitor remains muted.

Back to the problem originally described in this issue: NOT using the code to ignore changed on the `silenced` field, will re-introduce the exact problem described in my first post. Changes like this are shown when `terraform plan` is run:

```
  # module.golden-signals.datadog_downtime.error_log_downtime[0] will be updated in-place
  ~ resource "datadog_downtime" "error_log_downtime" {
      ~ active     = false -> true
        disabled   = false
      - end        = 1606225488 -> null
        id         = "22531760"
        message    = "This downtime has been created by Terraform."
        monitor_id = 443876
        scope      = [
            "*",
        ]
        start      = 1606225354
        timezone   = "UTC"
    }
```

**TL;DR** Using downtime resources in combination with this https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/monitor#silencing-by-hand-and-by-downtimes advise, makes it impossible to un-mute a monitor.

I hope my rather verbose description of the situation make sense, otherwise I am happy to rephrase.

Thanks,
Felix
